File: FRONTEND/src/service/eventService.js
import axios from "axios"

class EventService{

    constructor(){
        this.url = "http://localhost:8080/event";
    }

    getAllEvents(){
        return axios.get(this.url);
    }

    getEventById(id){
        return axios.get(this.url + "/id/" + id);
    }

    getEvent(date){
        return axios.get(this.url + "/" + date);
    }

    eventTable(middleDay){
        return axios.get(this.url + "/table/" + middleDay);
    }

    insertEvent(name,startDate,endDate){
        return axios.post(this.url,{name,startDate,endDate});
    }

    updateEvent(id,name,startDate,endDate){
        return axios.put(this.url,{id,name,startDate,endDate});
    }

    deleteEvent(id){
        return axios.delete(this.url + "/" + id)
    }

    getNames(){
        return axios.get(this.url + "/names");
    }

    getIdFromName(name){
        console.log(this.url + "/idFromName/" + name);
        return axios.get(this.url + "/idFromName/" + name);
    }

    getOrderedEvents(){
        return axios.get(this.url + "/ordered");
    }
}

export default new EventService();

File: BACKEND/src/main/java/com/example/BACKEND/service/interfaces/EventService.java
package com.example.BACKEND.service.interfaces;

import com.example.BACKEND.ObjectWrapper;
import com.example.BACKEND.dto.insert.EventInsertDto;
import com.example.BACKEND.dto.response.EventResponseDto;
import com.example.BACKEND.dto.update.EventUpdateDto;
import com.example.BACKEND.model.Event;
import javafx.util.Pair;
import org.springframework.stereotype.Service;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;


public interface EventService {

    ObjectWrapper<String,List<EventResponseDto>> getAllEvents();
    ObjectWrapper<String,EventResponseDto> getEventById(Long id);
    EventResponseDto getEvent(String dateTime);
    List<List<String>> eventTable(String middleDay);
    ObjectWrapper<List<String>,EventResponseDto> insertEvent(EventInsertDto eventInsertDto);
    ObjectWrapper<List<String>,EventResponseDto> updateEvent(EventUpdateDto eventUpdateDto);
    ObjectWrapper<String,EventResponseDto> deleteEvent(Long id);
    List<String> getNames();
    Optional<Long> findIdFromName(String name);
    List<Event> getOrderedEvents();
}
